pootle/apps/pootle_store/models.py

```diff
@@ -1972,7 +1971,7 @@ def update(self, overwrite=False, store=None, only_newer=False, user=None,
                                                        update_unitids, user,
                                                        revision=update_revision,
                                                        submission_type=submission_type)
-            self.file_mtime = disk_mtime
+            self.file_mtime = self.get_file_mtime() or self.file_mtime
```

Comment:
I'm not either sure I understand the purpose of the `or self.file_mtime` part. Independently of its meaning, I'm concerned it's a detail being taken care of in a ad-hoc fashion — easy to miss out.